src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java

```diff
@@ -373,270 +359,283 @@ default boolean isDamaged() {
      *
      * @param maxDamage The new max damage of the ItemStack
-     *
      * @docParam maxDamage 5
      */
     @ZenCodeType.Setter("maxDamage")
     default void setMaxDamage(int maxDamage) {
-        
+
         CraftTweakerAPI.apply(new ActionSetMaxDamage(this, maxDamage, this.getInternal().getMaxDamage()));
     }
-    
+
     /**
      * Returns the unlocalized Name of the Item in the ItemStack
      *
      * @return the unlocalized name.
      */
     @ZenCodeType.Getter("translationKey")
     default String getTranslationKey() {
-        
+
         return getInternal().getTranslationKey();
     }
-    
+
     /**
      * Sets the tag for the ItemStack.
      *
      * @param tag The tag to set.
-     *
      * @return This itemStack if it is mutable, a new one with the changed property otherwise
-     *
      * @docParam tag {Display: {lore: ["Hello"]}}
      */
     @ZenCodeType.Method
     IItemStack withTag(IData tag);
 
     /**
      * Removes the tag from this ItemStack.
+     *
      * @return This itemStack if it is mutable, a new one with the changed property otherwise
      */
     @ZenCodeType.Method
     IItemStack withoutTag();
-    
-    
+
+
     /**
      * Returns true if this ItemStack has a Tag
      *
      * @return true if tag is present.
      */
     @ZenCodeType.Getter("hasTag")
     default boolean hasTag() {
-        
+
         return getInternal().hasTag();
     }
-    
+
+    @Deprecated
+    default IData getTag() {
+
+        return NBTConverter.convert(getInternal().getTag());
+    }
+
     /**
      * Returns the NBT tag attached to this ItemStack.
      *
-     * @return IData of the ItemStack NBT Tag, null if it doesn't exist.
+     * @return MapData of the ItemStack NBT Tag, null if it doesn't exist.
      */
     @ZenCodeType.Getter("tag")
-    default IData getTag() {
-        
+    @ZenCodeType.Nullable
+    default MapData getTagNew() {
+
+        CompoundNBT tag = getInternal().getTag();
+        return tag == null ? null : new MapData(tag);
+    }
+
+    @Deprecated
+    default IData getOrCreateTag() {
+
+        if (getInternal().getTag() == null) {
+            getInternal().setTag(new CompoundNBT());
+        }
         return NBTConverter.convert(getInternal().getTag());
     }
-    
+
+
     /**
      * Returns the NBT tag attached to this ItemStack or makes a new tag.
      *
-     * @return IData of the ItemStack NBT Tag, empty tag if it doesn't exist.
+     * @return MapData of the ItemStack NBT Tag, empty tag if it doesn't exist.
      */
     @ZenCodeType.Getter("getOrCreate")
-    default IData getOrCreateTag() {
-        
-        if(getInternal().getTag() == null) {
+    default MapData getOrCreateTagNew() {
+
+        if (getInternal().getTag() == null) {
             getInternal().setTag(new CompoundNBT());
         }
-        return NBTConverter.convert(getInternal().getTag());
+        return new MapData(getInternal().getTag());
```

Comment:
If the suggestion of an overload is accepted, then you can just keep NBTConverter.convert here ^^